File: Coding Sample/Java/Fundamentals/Language/OOP/EnumTest.java
class EnumTest{

	/*
	interface RoomType{
		int ECONOMY = 1;
		int BUSINESS = 2;
		int EXECUTIVE = 3;
		int DELUXE = 4;
	}

	private static double payment(int days, int room){
		float rate = 0;
		switch(room){
			case RoomType.ECONOMY:
				rate = 525;
				break;
			case RoomType.BUSINESS:
				rate = 750;
				break;
			case RoomType.EXECUTIVE:
				rate = 850;
				break;
			default:
				rate = 975;
		}
		return 1.05 * rate * days;
	}
	*/

	enum RoomType{
		ECONOMY, BUSINESS, EXECUTIVE, DELUXE;
	}

	private static double payment(int days, RoomType room){
		float rate = 0;
		switch(room){
			case ECONOMY:
				rate = 525;
				break;
			case BUSINESS:
				rate = 750;
				break;
			case EXECUTIVE:
				rate = 850;
				break;
			default:
				rate = 975;
		}
		return 1.05 * rate * days;
	}

	public static void main(String[] args){
		int stay = Integer.parseInt(args[0]);
		System.out.printf("Payment for ECONOMY type room is %.2f%n", payment(stay, RoomType.ECONOMY));
		System.out.printf("Payment for BUSINESS type room is %.2f%n", payment(stay, RoomType.BUSINESS));
		System.out.printf("Payment for EXECUTIVE type room is %.2f%n", payment(stay, RoomType.EXECUTIVE));
		System.out.printf("Payment for DELUXE type room is %.2f%n", payment(stay, RoomType.DELUXE));
		//System.out.printf("Payment for CUSTOM type room is %.2f%n", payment(stay, 25));
	}
}

File: Coding Sample/Aassembly,UNIX, C/C/UDT/enumtest.c
#include <stdio.h>

enum RoomType {Economy=1, Business, Executive, Deluxe};

double HotelGetPayment(short stay, enum RoomType room)
{
	float rate;

	switch(room)
	{
	case Economy:
		rate = 975;
		break;
	case Business:
		rate = 1250;
		break;
	case Executive:
		rate = 2500;
		break;
	default:
		rate = 3750;
	}

	return 1.12 * stay * rate;
}

int main(void)
{
	short days;

	printf("Enter number of days: ");
	scanf("%hd", &days);

	printf("Total payment for economy class room will be %.2f\n", HotelGetPayment(days, Economy));
	printf("Total payment for business class room will be %.2f\n", HotelGetPayment(days, Business));
	printf("Total payment for executive class room will be %.2f\n", HotelGetPayment(days, Executive));
	printf("Total payment for deluxe class room will be %.2f\n", HotelGetPayment(days, Deluxe));
}


